lint/shellcheck.bzl

```diff
@@ -43,12 +43,19 @@ def shellcheck_action(ctx, executable, srcs, config, stdout, exit_code = None, o
     args.add_all(srcs)
     outputs = [stdout]
 
+    # Exit early when srcs is empty. This can happen when all srcs are generated files.
     if exit_code:
-        command = "{shellcheck} $@ >{stdout}; echo $? >" + exit_code.path
+        if len(srcs) == 0:
```

Comment:
this doesn't seem like a shellcheck-specific problem. What fix should we make to all the linters? They already have a filter_sources helper, so maybe instead of calling this function at all, they should just create the two dummy outputs (empty report and zero code)

Reply:
Makes sense. Some tools do exit successfully when no files are provided, but we could solve it for every tool in one place.
A dummy function that returns an empty report with the zero code should do it.